### PR TITLE
refactor(behavior_velocity_planner): new framework for creating virtual wall markers

### PR DESCRIPTION
**Pull request by maxime-clem**
## Description

Implement a new framework for managing virtual wall markers in the `behavior_velocity_planner`.
This simplifies the creation of the markers, moving the responsibility of  (1) assigning unique IDs and (2) creating `DELETE` markers from scene modules to scene managers.

| Before  | After |
| ------------- | ------------- |
| Each scene directly creates markers using the `VirtualWallMarkerCreator` class | Each scene creates `VirtualWall` objects and each manager creates the markers using the `VirtualWallMarkerCreator` class  |
| image | image |

## Tests performed

Test by visualizing virtual walls in the simple planning simulator.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by kyoichi-sugahara** [comment]
@maxime-clem 
Thank you so much!!
I will check the behavior.

File: common/motion_utils/src/marker/virtual_wall_marker_creator.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      virtual_wall.ns);
+    for (auto & marker : markers.markers) {
+      marker.id = marker_count_per_namespace[marker.ns].current++;
+      marker_array.markers.push_back(marker);
+    }
+  }
+  // create delete markers
+  visualization_msgs::msg::Marker marker;
+  marker.action = visualization_msgs::msg::Marker::DELETE;
+  for (const auto & [ns, count] : marker_count_per_namespace) {
```
Comment (kyoichi-sugahara) [inline common/motion_utils/src/marker/virtual_wall_marker_creator.cpp:76]: @maxime-clem 
If `marker_count_per_namespace` is zero, the loop may not be excuted and no action will be taken, correct?

Comment (maxime-clem) [reply common/motion_utils/src/marker/virtual_wall_marker_creator.cpp:76]: You mean if `marker_count_per_namespace.empty() == true` ? Then yes, this loop will not do anything.
Then for each namespace `ns`, we only create `DELETE` markers if `count.current < count.previous`, i.e., if we are publishing less `ADD` markers than before. For example, if at the previous iteration we published 3 markers with ids 1, 2, and 3 (`count.previous = 3`) and at the current iteration we only publish 1 (`count.current = 1`).

Comment (kyoichi-sugahara) [reply common/motion_utils/src/marker/virtual_wall_marker_creator.cpp:76]: Ok I understand.
Even if `marker_count.current == 0`, if `marker_count.previous` is not 0, namespace exists and `marker_count_per_namespace` doesn't become empty.
My concern was module unregistering remains final output `virtual_wall_marker`. But it not going to happen from your code.

**Review by kyoichi-sugahara — APPROVED** [review]
LGTM

**Review by soblin — APPROVED** [review]
I really appreciate for you PR ! And adding gtests is really nice !

**Review by TakaHoribe — APPROVED** [review]
LGTM!!! Thank you!

**Pull request merged**